modules/heat_conduction/test/tests/ad_heat_conduction/2drz_heated_rod_scaled.i

```diff
@@ -0,0 +1,111 @@
+# This test solves a 1D transient heat equation with a complicated thermal
+# conductivity in order to verify jacobian calculation via AD
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 20
+  ny = 20
+  xmax = 1
+  ymax = 1
+[]
+
+[Problem]
+  coord_type = RZ
+[]
+
+[Variables]
+  [./T]
+    initial_condition = 1.5
+  [../]
+[]
+
+[Kernels]
+  [./HeatDiff]
+    type = ADHeatConduction
+    variable = T
+    thermal_conductivity = thermal_conductivity
+    axis_scaling_vector = '1e3 1e2 0'
+  [../]
+  [./heat_dt]
+    type = ADHeatConductionTimeDerivative
+    variable = T
+    specific_heat = thermal_conductivity
+    density_name = thermal_conductivity
+  [../]
+  [./source]
+    type = ADMatHeatSource
+    variable = T
+    scalar = 1000
+  [../]
+[]
+
+[BCs]
+  [./top]
+    type = ADFunctionPresetBC
+    variable = T
+    boundary = top
+    function = 't+1'
+  [../]
+  [./bottom]
+    type = ADPresetBC
+    variable = T
+    boundary = bottom
+    value = 2
+  [../]
+
+  [./left]
+    type = ADPresetBC
+    variable = T
+    boundary = left
+    value = 1
+  [../]
+  [./right]
+    type = ADFunctionPresetBC
+    variable = T
+    boundary = right
+    function = 't*2+1'
+  [../]
+[]
+
+[Materials]
+  [./k]
+    type = ADThermalConductivityTest
+    c = T
+    temperature = T
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = NEWTON
+  num_steps = 2
+  dt = 1
+  automatic_scaling = true
```

Comment:
@bwspenc You can see the effect of geometric scaling with this input file:

`automatic_scaling=true` -> Nonlinear |R| = 2.035955e-08
`automatic_scaling=false`, -> Nonlinear |R| = 3.682260e-01, which is probably fine
`automatic_scaling=false`, reference residual problem -> problem does not converge in the second time step, and cuts the time.

The non scaled input file works fine with/without scaling, and with/without reference residual. There are probably details here that are being glossed over, but it's unsurprising the anisotropic diffusion parameter results in numerical difficulties, but maybe @lindsayad can shed light on how residual scaling helps with a single variable, but in 2D problems.

Reply:
For background @lindsayad, @bwspenc and I were talking about geometric scaling, and Ben was surprised that residual scaling would help in highly anisotropic diffusion coefficients in 2D, as he was thinking that the residual scaling is constant for each variable. I was assuming that the variable is scaled by dimension as well...can you clarify?

Reply:
> but maybe @lindsayad can shed light on how residual scaling helps with a single variable, but in 2D problems.

Huh?

> as he was thinking that the residual scaling is constant for each variable. I was assuming that the variable is scaled by dimension as well...can you clarify?

@bwspenc is correct. There's only one scaling factor for each non-linear variable. So if we were doing mechanics there would be different scaling factors for `disp_x` and `disp_y` but that is it as far as dimensions go. Here since the title is heat conduction, I'm going to assume we're generally only working with `T` so there should be only one scaling factor regardless of whether we are in 1, 2, or 3D. The automatic scaling calculation is really quite simple...it looks for the largest magnitude diagonal entry for a variable in the preconditioning matrix and then the scaling factor is set to the the reciprocal of that number. If you're seeing tiny initial residuals after automatic scaling, then there would seem to be a very large discrepancy between the magnitude of your residuals and Jacobians which could be due to an initial condition I suppose.

Reply:
This was my misunderstanding of scaling. The reason automatic scaling worked here is because it scaled the residual, and helped the relative tolerance to be reached.